pkg/webhooks/server.go

```diff
@@ -360,7 +374,61 @@ func (ws *WebhookServer) resourceMutation(request *v1beta1.AdmissionRequest) *v1
 
 func (ws *WebhookServer) resourceValidation(request *v1beta1.AdmissionRequest) *v1beta1.AdmissionResponse {
 	logger := ws.log.WithName("resourceValidation").WithValues("uid", request.UID, "kind", request.Kind.Kind, "namespace", request.Namespace, "name", request.Name, "operation", request.Operation)
+	checked, err := userinfo.IsRoleAuthorize(ws.rbLister, ws.crbLister, ws.rLister, ws.crLister, request)
+	if err != nil {
+		logger.Error(err, "failed to get RBAC infromation for request")
+	}
+
+	if !checked {
+		if request.Operation == v1beta1.Delete || request.Operation == v1beta1.Update {
```

Comment:
We can check the operation first, then decide whether needs to get roles `userinfo.IsRoleAuthorize`, in this way we can return non-targeted requests faster.

Reply:
ok